src/librustc_driver/driver.rs

```diff
@@ -177,7 +180,7 @@ pub fn compile_input(sess: &Session,
                                     &arena,
                                     &arenas,
                                     &crate_name,
-                                    |tcx, analysis, incremental_hashes_map, result| {
+                                    |tcx, analysis, incremental_hashes_map, borrow_analysis_map, result| {
```

Comment:
Yeah, I am unkeen on threading this around.  I think we should just modify the save-analysis code to repeat the `borrowck` queries -- though this implies that the results will be stored in the memoization maps.